sync/sync-impl/src/main/java/com/duckduckgo/sync/impl/SyncServiceRemote.kt

```diff
@@ -232,22 +231,26 @@ class SyncServiceRemote @Inject constructor(private val syncService: SyncService
         }
 
         return onSuccess(response) {
-            Result.Success(response.body()!!)
+            Result.Success(response.body())
         }
     }
 
-    override fun getAllData(token: String): Result<SyncDataResponse> {
+    override fun getBookmarks(token: String, since: String): Result<SyncDataResponse> {
         val response = runCatching {
-            val patchCall = syncService.data("Bearer $token")
+            val patchCall = if (since.isNotEmpty()) {
+                syncService.bookmarksSince("Bearer $token", since)
```

Comment:
discussed offline, the API doesn’t allow for the use of get without sending a since property per feature. we don’t have other features now, so there’s no point in adding extra logic now.